### DISCUSSION TITLE
Regarding b-vectors

### DISCUSSION THREAD

**Discussion by gianfrancortes** [dipy/dipy#2948] [Preprocessing] [answered]
Hi DIPY!

What space do the b-vectors live in? In other words, the components provided in the .bvec file are with respect to what coordinate frame?

Many thanks.

**Comment by cookpa** [comment]
I have the same question, please feel free to point me to a documentation page if this is answered somewhere.

If I understand correctly, BIDS requires bvecs to be in "RADIOLOGICAL" orientation (per fslorient), as required by FSL. But as far as I know, it does not require images to have the same convention. So we can be in a situation where the bvecs are not in the same index space as the image (ie, a NEUROLOGICAL DWI image).

What does dipy do internally? Does BIDS data ever need the bvecs to be modified on import?

Thanks in advance

**Reply by skoudoro** [reply]
tagging @arokem for this one

**Reply by skoudoro** [reply]
I still need confirmation from @arokem but I believe we do like bids and FSL. So b-vectors live in subject space.

**Comment by arokem** [comment, marked as answer]
I think that the answer here is that we require absolutely nothing from the b-vectors. The only thing we _assume_ is that the data and the b-vectors are _in the same space_. That is, if your b-vecs are oriented in a way that is different than your data (e.g., radiological vs. neurological), the preprocessing pipeline that precedes DIPY needs to correct that. Does that answer the question, or am I missing something more subtle?

**Reply by cookpa** [reply]
Thanks for this. I think that covers my question.

**Reply by skoudoro** [reply]
related issue: https://github.com/dipy/dipy/issues/1047#issuecomment-1819486217

We need to document it
